node/src/components/linear_chain_sync/operations.rs

```diff
@@ -91,6 +91,26 @@ where
     }
 }
 
+async fn fetch_trie_retry_forever(
+    effect_builder: EffectBuilder<JoinerEvent>,
+    id: Digest,
+) -> FetchedData<Trie<Key, StoredValue>, NodeId> {
+    loop {
+        let peers = effect_builder.get_peers_in_random_order::<NodeId>().await;
+        trace!(?id, "attempting to fetch a trie",);
+        match effect_builder.fetch_trie(id, peers).await {
+            Ok(fetched_data) => {
+                trace!(?id, "got trie successfully",);
+                return fetched_data;
+            }
+            Err(error) => {
+                warn!(?id, %error, "fast sync could not fetch a trie; trying again")
```

Comment:
Ah, there it is! If the error is really that we couldn't deserialize a cryptographically valid trie we should give up.